bones/textBone.py

```diff
@@ -7,108 +7,193 @@
 from server.config import conf
 from google.appengine.api import search
 from server.bones.stringBone import LanguageWrapper
-import logging
+import logging, string
 
-_attrsMargins = ["margin","margin-left","margin-right","margin-top","margin-bottom"]
-_attrsSpacing = ["spacing","spacing-left","spacing-right","spacing-top","spacing-bottom"]
-_attrsDescr = ["title","alt"]
 _defaultTags = {
-	"validTags": [	'font','b', 'a', 'i', 'u', 'span', 'div','p', 'img', 'ol', 'ul','li','acronym', #List of HTML-Tags which are valid
-				'h1','h2','h3','h4','h5','h6', 'table', 'tr', 'td', 'th', 'br', 'hr', 'strong'],
-	"validAttrs": {	"font": ["color"], #Mapping of valid parameters for each tag (if a tag is not listed here: no parameters allowed)
-					"a": ["href","target"]+_attrsDescr,
-					"acronym": ["title"],
-					"div": ["align","width","height"]+_attrsMargins+_attrsSpacing,
-					"p":["align","width","height"]+_attrsMargins+_attrsSpacing,
-					"span":["align","width","height"]+_attrsMargins+_attrsSpacing,
-					"img":[ "src","target", "width","height", "align" ]+_attrsDescr+_attrsMargins+_attrsSpacing,
-					"table": [ "width","align", "border", "cellspacing", "cellpadding" ]+_attrsDescr,
-					"td" : [ "colspan", "rowspan", "width", "height" ]+_attrsMargins+_attrsSpacing
-				},
-	"validStyles": ["font-weight","font-style","text-decoration","color", "display"], #List of CSS-Directives we allow
-	"singleTags": ["br","img", "hr"] # List of tags, which dont have a corresponding end tag
+	"validTags": [  # List of HTML-Tags which are valid
+		'b', 'a', 'i', 'u', 'span', 'div', 'p', 'img', 'ol', 'ul', 'li', 'abbr', 'sub', 'sup',
+		'h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'table', 'thead', 'tbody', 'tfoot', 'tr', 'td', 'th', 'br',
+		'hr', 'strong', 'blockquote', 'em'],
+	"validAttrs": {  # Mapping of valid parameters for each tag (if a tag is not listed here: no parameters allowed)
+		"a": ["href", "target", "title"],
```

Comment:
I would like to see 'rel' in validAttrs for a.

target="blank" should be used in conjunction with rel="noopener noreferrer".

Here more into the subject:
1. https://developers.google.com/web/tools/lighthouse/audits/noopener
2. https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/

Reply:
That's enforced automatically if target is set to blank (Line 152f)